core/tests/test_utils.py

```diff
@@ -622,6 +622,9 @@ def process_and_flush_pending_tasks(self, queue_name=None):
                 if task.url == '/_ah/queue/deferred':
                     from google.appengine.ext import deferred
                     deferred.run(task.payload)
+                elif feconf.EMAILS_TASK_PREFIX in task.url:
+                    # Flush task if task is of emails taskqueue.
```

Comment:
Mm, that's not so good, it introduces skew, and it's quite implicit. This sort of thing tends to lead to problems down the road.

Why can't you use a payload, like the other tasks do? Then feed in params similar to what is done in the 'else' clause. Otherwise, your tests are kind of bypassing the task queue.

Also, why do you need the user_id parameter in the request?

Basically, I'm trying to understand what makes this fundamentally different from the mapreduce tasks in the 'else' clause. It would definitely be better to do something similar here, if possible.